src/Symfony/Component/Security/Core/Authorization/AccessDecisionManagerInterface.php

```diff
@@ -17,16 +17,20 @@
  * AccessDecisionManagerInterface makes authorization decisions.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @method AccessDecision getDecision(TokenInterface $token, array $attributes, object $object = null)
  */
 interface AccessDecisionManagerInterface
 {
     /**
      * Decides whether the access is possible or not.
      *
-     * @param array $attributes An array of attributes associated with the method being invoked
-     * @param mixed $object     The object to secure
+     * @param array  $attributes An array of attributes associated with the method being invoked
+     * @param object $object     The object to secure
+     *
+     * @return bool true if the access is granted, false otherwise
      *
-     * @return bool
+     * @deprecated since 5.3, use {@see getDecision()} instead.
      */
-    public function decide(TokenInterface $token, array $attributes, $object = null);
+    public function decide(TokenInterface $token, array $attributes, $object = null): bool;
```

Comment:
Adding the return type in an interface is a BC break. That's a no-go (especially to deprecate it)